File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        for (int &t : nums) {
            mp[t]++;
        }
        
        // Index represents frequency, value is list of numbers with that frequency
        vector<vector<int>> buckets(nums.size() + 1);
        for (auto &[num, count] : mp) {
            buckets[count].push_back(num);
        }
        
        vector<int> ans;
        // Iterate backwards from highest possible frequency
        for (int i = buckets.size() - 1; i >= 0 && ans.size() < k; --i) {
            for (int num : buckets[i]) {
                ans.push_back(num);
                if (ans.size() == k) break;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:

        ans = [[]]
        for i in nums:
            
            new_subset = []
            for curr in ans:
                new_subset.append(curr + [i])
            
            ans.extend(new_subset)
            

        
        return ans ;
        